plugins/useractivity/useractivity.go

```diff
@@ -103,22 +110,35 @@ func (r *reporter) submitter() {
 		// channel, and the loop body contains a Remove call (which acquires the
 		// shard's mutex), we deadlock.
 		//
+		tickTime := time.Now()
 		toRemove := []string{}
```

Comment:
Minor optimization: since submitter is a single goroutine, we can reduce allocation by declaring this outside the loop and using 

`toRemove = toRemove[:0]`

to clear it here.